garden-service/src/process.ts

```
@@ -144,7 +144,9 @@ export async function processModules(
 
   // Experimental HTTP API and dashboard server.
   if (process.env.GARDEN_ENABLE_SERVER === "1") {
-    await startServer(garden, log)
+    // allow overriding automatic port picking
+    const port = Number(process.env.GARDEN_SERVER_PORT) || undefined
```

Should we validate that `process.env.GARDEN_SERVER_PORT` is a valid (port) number? `Number` just returns `NaN` if the parameter is not a number.

---

It will default to the random port if it isn't valid and it will throw and error if it can't bind to the port so I don't think it's nessary.